XenAdmin/TabPages/SnapshotsPage.cs

```diff
@@ -426,16 +426,7 @@ private IList<VM> RefreshDataGridView(IList<VM> snapshots)
             for (int i = 0; i < snapshots.Count; i++)
             {
                 VM snapshot = snapshots[i];
-                if (!(snapshot.is_snapshot_from_vmpp && !toolStripMenuItemScheduledSnapshots.Checked))
-                {
-                    snapshot.PropertyChanged -= snapshot_PropertyChanged;
-                    snapshot.PropertyChanged += snapshot_PropertyChanged;
-                    //Build DataGridView
-                    SnapshotDataGridViewRow row = new SnapshotDataGridViewRow(snapshot);
-                    row.Tag = snapshot;
-                    DataGridView.Rows.Add(row);
-                }
-                else if (!(snapshot.is_vmss_snapshot && !toolStripMenuItemScheduledSnapshots.Checked))
+                if (!((snapshot.is_snapshot_from_vmpp || snapshot.is_vmss_snapshot) && !toolStripMenuItemScheduledSnapshots.Checked))

```

Comment:
oh alright, i had misunderstood the interpretation of checkbox, will update the PR. Thanks for your suggestion.